lib/src/controller/click_handler.dart

```diff
@@ -0,0 +1,45 @@
+import 'package:link_five/src/logic/actions/move_tile_action.dart';
+import 'package:link_five/src/logic/actions/place_tile_action.dart';
+import 'package:link_five/src/logic/game_action.dart';
+import 'package:link_five/src/model/game/game_state.dart';
+import 'package:link_five/src/model/game/player_color.dart';
+import 'package:link_five/src/model/game/tile_location.dart';
+
+class ClickHandler {
+  TileLocation? source;
+  GameAction? handleGameClick(
+      GameState state, TileLocation location, PlayerColor playerColor) {
+    if (state.tilesAvailable) {
+      return PlaceTileAction(
+        playerColor: playerColor,
+        location: location,
+      );
+    } else {
+      final colorAtLocation = state.gameBoard[location]?.color;
+      print(colorAtLocation);
+      if (source == null) {
+        if (colorAtLocation != null && colorAtLocation == playerColor) {
```

Comment:
No need to `null` check here. If `colorAtLocation` is `null`, `colorAtLocation == playerColor` will always evaluate to false since `playerColor` is not nullable.
```suggestion
        if (colorAtLocation == playerColor) {
```